src/features/cosoul/api/cosoul.ts

```diff
@@ -51,17 +72,10 @@ export const getOnChainPGIVE = async (tokenId: number) => {
 };
 
 // set the on-chain PGIVE balance for a given token
-export const setOnChainPGIVE = async (tokenId: number, amt: number) => {
```

Comment:
make two functions, one for multi setting, one for single setting

Reply:
@crabsinger  done. we have now two functions and the tests are updated accordingly